### ISSUE TITLE
No pull synchro with the marlin repo

### ISSUE THREAD

**Issue by axl12ap** [akaJes/marlin-config#84]
Good morning,
with the latest version of marlin-conf 2.10.7 (npm package 2.10.5) I've a problem when I try to get the latest marlin repo version.

Opening marlin-conf, tell a new folder to get all sourcers from Git, waiting that program pull all sources.
Then I switch branches "origin/bugfix-2.0.x" and wait for the updates of source code (tried also to update with UPDATE button).

If then I compare the folder marlin-conf has downloaded and a ZIP zip version of latest marlin Git  repo, there're a lot of difference in files.

Please check it.

**Comment by akaJes** [comment]
The app uses `git fetch` command and it updates only the repository not
working folder, to do it would use 'reset' menu command
https://i.stack.imgur.com/nWYnQ.png

Пт, 19 жовт. 2018, 11:58 користувач Alessandro <notifications@github.com>
пише:

**Comment by axl12ap** [comment]
Sorry, I don't understood, please explain me the correct sequence to have the working "bugfix-2.0.x" folder on my local drive with marlin-conf

**Comment by yet-another-average-joe** [comment]
Hello,

This is not an issue, it's a feature ! And this is not a forum, it is intended for bug reports and feature requests. But I can try to explain... (a forum dedicated to marlin-conf would be great !)

As I understand Marlin-Conf, it is not intended to be used the way you attempt to use it ! You cannot change version (clone repository) unless you cancel all changes. This is what akaJes explained : Reset command. Switching versions from the menu would probably be a mess ! Imagine switching from a newer Marlin version  to an older one : lots of #define would be lost !

So, if you want to keep your current version untouched :

Create a new folder for the latest Marlin 2.x bugfix. Then open this foder with Marlin-Conf. It will clone the last 1.1x repository. Then, change for branch 2.x. You will get a vanila 2.x (last bugfix)

Now, compare config*.h versions (open files and search for CONFIGURATION_H_VERSION and CONFIGURATION_ADV_H_VERSION). If versions are the same, you can copy old config*.h files to the newly created folder, overwriting the vanila ones. Don't forget other files if needed : _Bootscreen.h, _Statusscreen.h.

If version numbers differ, drag the files in marlin-conf window, and they will be upgraded. You can also do this if headers versions are the same, they will remain the same (to be verified, I did'nt !)

After you do a "drag/drop upgrade", it is a good idea to open 2 marlin-conf instances, one for the old Marlin version and one for the new one. Then, compare the values. Report any problem to akaJes.

You can also cancel all changes (backup your config*.h files !) : Reset. This is what akaJes explained to you ! Then restore your files (if versions are the same), or simply drag/drop. On my side, I prefer keeping multiple versions, each one in its own directory. It's easier to switch (and flash) from one version to another.

**Comment by robustini** [comment]
When i try to change branch from 1.1.x to 2.0 bug-fix i've a lot of error, i don't want to commit anything, i'm not a Marlin developer, i want just switch to the other branch with my settings... this is not intuitive in your configurator.

![immagine](https://user-images.githubusercontent.com/3864712/47705439-dd08a780-dc26-11e8-91c4-02d730fd5e9e.png)

**Comment by yet-another-average-joe** [comment]
These errors probably mean that you just opened a firmware folder with missing example files. It is easy to reproduce. This is NOT marlin-conf related, but Git related ! Marlin-conf works on Git repositories, not directly on Marlin sources downloaded from a vendor's website. Git needs a very specific folders hierarchy. The menu is intended to clone a repo, not to upgrade a firmware : Change, Branches, Update, and Reset are the Git command wrappers.

Marlin-conf is a Git wrapper with specialized editors + a plaformio wrapper + collaborative tools. It is some sort of integrated development environment (IDE). It is intuitive, but you must have some basic understanding of Git (a developpers tool !) : 

1 - create an EMPTY  folder where Git will clone repository
2 - launch marlin-conf and open this new folder : the last 1.9.x vanila repo will be cloned by default (= last stable version)
3 - change version or branch using the menu : the folder now contains the 2.0 vanila repo
4 - cutomize this new Marlin repo by editing parameters in conf*.h tabs, or drag/drop your conf*.h files in marlin-conf window ; be aware that a vendor could have changed some other sources, you have to verify this point !!!
5 - verify parameters (some bugs have been pointed out)
5 - compile and flash, directly or using the hex file (.pioenvs/[motherboard]/firmware.hex) ; the hex file is needed for example if you want to flash through OctoPrint over the network (with the dedicated plugin)

**Comment by robustini** [comment]
Where is the marlin-conf cfg file? I want to erase all the customized directory list and start clean...

**Comment by yet-another-average-joe** [comment]
Not sure what you are asking for... What customized directory list ? The examples ?

Doing so, you are just killing the Git repo, and you will get only errors ! Do NOT do this.

**Comment by robustini** [comment]
I mean "how to erase the custom directory list" from the configurator.
I've two custom directory but only one good...

![immagine](https://user-images.githubusercontent.com/3864712/47718724-63ce7c00-dc49-11e8-9140-dd103764467e.png)

**Comment by akaJes** [comment]
`~/.data-store/marlin-config.json`
i belive that you know about the `~` is the user's home folder?

**Comment by akaJes** [comment]
@robustini before version changing or updating changes from branch it must be resetted from RESET menu command

**Comment by yet-another-average-joe** [comment]
On Windows OS, "~/.data-store" becomes "[system disk]\users\[current user]\data-store", but this is not really usefull, as the "new" button and the MRU list do nothing when clicking, Furthermore, the viewport is too small to display full paths. The "open folder" button remembers the last opened folder, and this is OK.

**Comment by akaJes** [comment]
The MRU has popups for each row with full path, maybe a windows has some issue with displaying?

**Comment by yet-another-average-joe** [comment]
Yes, as showned by the image posted by Robustini, the viewport is too small, and the paths are not cliclable (they do nothing).

[EDIT] there is not popup !

This is not really an issue. It is just cosmetic.

Under Windows at least, the MRU is not that usefull. When clicking the button "open folder", it opens an OpenFileDialog pointing to the last opened firmware, and everything can be done from there : browsing, creating a new folder, renaming, etc. The user will probably create all his working folders as a subfolder of a root folder, and, after cloning, rename each and every firmware (the "Marlin" folder created by Git).
